src/Client/widgets/Task.fs

```diff
@@ -10,157 +10,226 @@ open Fable.FontAwesome
 open Fable.FontAwesome.Free
 open Fulma
 open Markup
+open Utils
+
+[<Literal>] 
+let DefaultWord = ""
+let DefaultAnswers = [||]
 
 type Task = { 
     Word : string 
     Answers : string[]
 }
 
+type InputState = 
+    | Unknown
+    | Wrong
+    | Right
+    | Shown
+
+type Input = string
+type TaskWithInput = Task * Input * InputState
+
+type State = 
```

Comment:
I decided to go with a finite automaton approach, we have 3 high level states and 4 sub-states in case when use typed something.

Reply:
I think that's a really good idea actually, perfectly fitting the function paradigm